flow/paint_utils.cc

```diff
@@ -24,14 +24,19 @@ sk_sp<SkShader> CreateCheckerboardShader(SkColor c1, SkColor c2, int size) {
                        SkSamplingOptions());
 }
 
-void DrawCheckerboard(SkCanvas* canvas, SkColor c1, SkColor c2, int size) {
-  SkPaint paint;
-  paint.setShader(CreateCheckerboardShader(c1, c2, size));
-  canvas->drawPaint(paint);
-}
-
 }  // anonymous namespace
 
+void DrawCheckerboard(SkCanvas* canvas,
+                      DisplayListBuilder* builder,
+                      const SkRect& rect) {
+  if (canvas) {
+    DrawCheckerboard(canvas, rect);
+  }
+  if (builder) {
```

Comment:
else if?

Reply:
That may be true, and true of any number of methods in this file, but very soon one of these variations will be eliminated entirely so I'd like to leave the lines separated. Originally I allowed both until I realized that we should only be updating a single delegate at a time.

Perhaps in another pass I could switch these to use a DelegateHandler with 3 variants for SkCanvas, DLBuilder, and MutatorsStack...